### DISCUSSION TITLE
how to split UdpSocket?

### DISCUSSION THREAD

**Discussion by ta3pks** [tokio-rs/tokio#3755] [Q&A] [answered]
UdpSocket used to have a split method which doesnt exist anymore, How does one split a udp socket at the moment?

**Comment by Darksonn** [comment, marked as answer]
This is explained in the documentation for [`UdpSocket`][1].

> This type does not provide a `split` method, because this functionality can be achieved by instead wrapping the socket in an [`Arc`][2]. Note that you do not need a `Mutex` to share the `UdpSocket` — an `Arc<UdpSocket>` is enough. This is because all of the methods take `&self` instead of `&mut self`. Once you have wrapped it in an `Arc`, you can call `.clone()` on the `Arc<UdpSocket>` to get multiple shared handles to the same socket. An example of such usage can be found further down.

There is also an example in the documentation:
```Rust
use tokio::{net::UdpSocket, sync::mpsc};
use std::{io, net::SocketAddr, sync::Arc};

#[tokio::main]
async fn main() -> io::Result<()> {
    let sock = UdpSocket::bind("0.0.0.0:8080".parse::<SocketAddr>().unwrap()).await?;
    let r = Arc::new(sock);
    let s = r.clone();
    let (tx, mut rx) = mpsc::channel::<(Vec<u8>, SocketAddr)>(1_000);

    tokio::spawn(async move {
        while let Some((bytes, addr)) = rx.recv().await {
            let len = s.send_to(&bytes, &addr).await.unwrap();
            println!("{:?} bytes sent", len);
        }
    });

    let mut buf = [0; 1024];
    loop {
        let (len, addr) = r.recv_from(&mut buf).await?;
        println!("{:?} bytes received from {:?}", len, addr);
        tx.send((buf[..len].to_vec(), addr)).await.unwrap();
    }
}
```

[1]: https://docs.rs/tokio/1/tokio/net/struct.UdpSocket.html
[2]: https://doc.rust-lang.org/stable/std/sync/struct.Arc.html
